### PR TITLE
Add sanity checks for removed component image variables

### PR DESCRIPTION
**Pull request by michaelgugino**
This commit adds failure condition if any removed variables are found.

### CODE REVIEW

File: roles/lib_utils/action_plugins/sanity_checks.py [hunk 1]
```diff
@@ -54,6 +92,21 @@ def to_bool(var_to_check):
...
+    """Fails if removed variables are found"""
+    found_removed = []
+    for item in REMOVED_VARIABLES:
+        if item in hostvars[host]:
+            found_removed.append(item)
+
+    if found_removed:
+        msg = "Found removed variables: "
+        for item in found_removed:
+            msg += "{} is replaced by {}; ".format(item[0], item[1])
```
Comment (sdodson) [inline roles/lib_utils/action_plugins/sanity_checks.py]: Can we make it clear that the replacement variable is the fully qualified reference to the image?

Comment (michaelgugino) [reply roles/lib_utils/action_plugins/sanity_checks.py]: I want to add a link to some documentation page with this error message.  We need a page that I can link to.

Comment (sdodson) [reply roles/lib_utils/action_plugins/sanity_checks.py]: Ok, we can follow up on that.

**Comment by sdodson** [comment]
BTW, proposed documentation structure for this change


Based on oreg_url='registry.example.com/ocp/ose-${component}:${version}' these are the images and relevant variables. Note that if you're overriding any of the defaults you need to specify the entire pull spec, no attempt to parse this value will be made.

| Old variable(s) | New image(s) based on oreg_url | Override variable |
|-----------------|-----------------------------------------|------------------------|
| openshift_metrics_image_prefix, openshift_metrics_image_version | registry.example.com/ocp/ose-metrics-hawkular:${version}, registry.example.com/ocp-metrics-cassandra:${version} | openshift_metrics_image_prefix |

File: roles/lib_utils/action_plugins/sanity_checks.py [hunk 2]
```diff
@@ -43,6 +43,46 @@
     'openshift_prometheus_alertmanager_storage_kind',
     'openshift_prometheus_storage_kind')
 
+REMOVED_VARIABLES = (
+    # TODO(michaelgugino): Remove these in 3.11
+    ('openshift_metrics_image_prefix', 'openshift_metrics_<component>_image'),
+    ('openshift_metrics_image_version', 'openshift_metrics_<component>_image'),
+    ('openshift_grafana_proxy_image_prefix', 'openshift_grafana_proxy_image'),
+    ('openshift_grafana_proxy_image_version', 'openshift_grafana_proxy_image'),
+    ('openshift_logging_image_prefix', 'openshift_logging_image'),
```
Comment (sdodson) [inline roles/lib_utils/action_plugins/sanity_checks.py]: I don't think openshift_logging_image is anything?

Comment (michaelgugino) [reply roles/lib_utils/action_plugins/sanity_checks.py]: Needed for /home/mgugino/git/openshift-ansible/roles/openshift_logging/templates/jks_pod.j2

**Pull request merged**